cherno/guider.py

```diff
@@ -192,171 +200,127 @@ def __init__(
-            fit_focus=fit_focus,
-        )
+        if self.solve_pointing.offset_dec != full_offset[1]:
+            return False
 
-        if (
-            self.target_rms is not None
-            and ast_solution.valid_solution
-            and ast_solution.rms > 0
-            and ast_solution.rms <= self.target_rms
-            and stop_at_target_rms
-        ):
-            self.command.warning("RMS has been reached. Not applying correction.")
-            correct = False
+        if self.solve_pointing.offset_pa != full_offset[2]:
+            return False
 
-        if correct and ast_solution.valid_solution is True:
-            await self.correct(
-                ast_solution,
-                full=full_correction,
-                wait_for_correction=wait_for_correction,
-                apply_focus=fit_focus,
-            )
-        else:
-            self.command.info(
-                acquisition_valid=ast_solution.valid_solution,
-                did_correct=any(ast_solution.correction_applied),
-                correction_applied=ast_solution.correction_applied,
-            )
+        return True
 
-        if self.command.actor:
-            update_proc_headers(ast_solution, self.command.actor.state)
+    def check_wcs(self, guide_data: list[GuideData]):
+        guide_cameras = self.command.actor.state.guide_cameras
+        wcs_solved = [
+            ad.camera
+            for ad in guide_data
+            if ad.solved is True and ad.camera in guide_cameras
+        ]
+        if len(wcs_solved) > 1:
+            # if 2 or more cameras have wcs solns, use "coordio"
+            for gd in guide_data:
+                gd.solved = True
+                gd.solve_method = "coordio"
 
-        return ast_solution
+        return wcs_solved
 
-    async def fit(
+    def try_rapid_solve(
         self,
-        data: list[GuideData],
-        offset: list[float] | None = None,
-        scale_rms: bool = False,
-        fit_focus: bool = True,
-        only_radec: bool = False,
-        auto_radec_min: int = 2,
-        fit_all_detections: bool = True,
+        guide_data: list[GuideData],
+        ext_data: list[ExtractionData],
+        offset: list[float] | None,
     ):
-        """Calculate the astrometric solution."""
+        sp_guider_fit = None
+        converged = self.check_convergence(ext_data, offset)
+
+        if converged:
+            dfList = []
+            for ex in ext_data:
+                regions = ex.regions.copy().reset_index(drop=True)
+                gfaNum = int(ex.camera.strip("gfa"))
+                regions["gfaNum"] = gfaNum
+                gain = ex.gain
+                regions["gain"] = gain
+                dfList.append(regions)
+            df = pandas.concat(dfList).reset_index(drop=True)
+
+            sp_guider_fit = self.solve_pointing.reSolve(
+                ex.exposure_no, ex.obstime, ex.exptime, df
+            )
+            if self.solve_pointing.guide_rms_sky > 1:
```

Comment:
This would fail if `self.solve_pointing` is `None`, which is the default value.

Reply:
again like the rest of the code in this block, a check that solve_pointing is not None is already in place